File: Palindrome Number/Palindrome_Number.py
from math import log10


class Solution:
    def isPalindrome(self, x: int) -> bool:
        return False if x < 0 else str(x) == str(x)[::-1]


class LogarithmSolution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        if 0 < x < 10:
            return True

        digit_count = 1 + int(log10(x))
        half = int(digit_count / 2)
        digits = list(map(lambda exp: int(x / pow(10, exp) % 10), range(digit_count)))
        left = digits[:half]
        digits.reverse()
        right = digits[:half]
        return all(map(lambda y: y[0] == y[-1], zip(left, right)))

File: Palindrome Number/palindrome-number.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        bool ret = false;
        vector<int> vec;
        int num = x;
        double reverse = 0;
        if (x >= 10)
        {
            while (num >= 10)
            {
                vec.push_back(num % 10);
                num = num / 10;
                if (num < 10)
                    vec.push_back(num);
            }

            int p = vec.size();
            for (int i = 0; i < vec.size(); i++)
            {
                reverse += (vec[i] * pow(10, p - 1));
                if (reverse > INT_MAX) return false;
                p--;
            }
        }

        if ((x > 0 and x < 10) || reverse == x)
            ret = true;

        return ret;
    }
};

int main()
{
    Solution s;
    bool b = s.isPalindrome(1234567899);
    return 0;
}